Add Kb - Recurring job does not create new jobs after detach and attach (#290)
longhorn/website: content/kb/troubleshooting-recurring-job-stuck-after-detach-attach.md

---

> title: "Troubleshooting: Recurring job does not create new jobs after detach and attach"

Author: Fixed.

> title: "Troubleshooting: Recurring job does not create new jobs after detach and attach"

Author: Fixed.

author: Chin-Ya Huang
draft: false
date: 2021-04-21
categories:
  - "recurring job"
---

## Applicable versions

> All Longhorn versions.

Reviewer: Should we indicate `Version v1.1.1 and earlier` ? So, we fix this in v1.1.2 we don't have to modify the KB

Author: I have no problem having the version pre-set before the fix, but thought it might make more sense to be in sync with the current status. @innobead , can we get some opinion on this?

Reviewer 2: I think we can revisit back later when we really fix the issue. @PhanLe1010, update this action item in the related ticket? Thanks.

## Symptoms

> Recurring job does not create new jobs when the volume is attached after being detached for sometimes.

Reviewer: ```suggestion
Recurring job does not create new jobs when the volume is attached after being detached for a long time.
```

Author: Thanks, addressed.

According to Kubernetes [CronJob limitations](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-job-limitations):
> For every CronJob, the CronJob Controller checks how many schedules it missed in the duration from its last scheduled time until now. If there are more than 100 missed schedules, then it does not start the job and logs the error
>
>```
>Cannot determine if job needs to be started. Too many missed start time (> 100). Set or decrease .spec.startingDeadlineSeconds or check clock skew.
>```

> The means duration of the attach/detach that the recurring job can tolerate is depending on the scheduled interval.

Reviewer 2: ```suggestion
That means the duration of the attach/detach operation that the recurring job can tolerate is depending on the scheduled interval.
```

Author: Fixed.

> For example, If the recurring backup job is set to run every minute, then the toleration would be 100 minutes.

Reviewer 2: ```suggestion
For example, if the recurring backup job is set to run every minute, then the toleration would be 100 minutes.
```

Author: Fixed.
